packages/frontend/src/screens/app/screens/tournaments/screens/tournament/screens/match/match-view.tsx

```diff
@@ -14,24 +16,54 @@ export interface MatchViewMatch {
 	secondTeam: Team;
 	winner: Team | null;
 	contested: boolean;
+	tournament: { id: string; owner: { id: string } };
+	finalized: boolean;
 }
 
 interface Props {
 	reportWin: (teamId: string) => void;
 	reportContested: (contested: boolean) => void;
+	reportFinalized: () => void;
 	match: MatchViewMatch;
 	loading: boolean;
+	tournament: {
+		id: string;
+		owner: { id: string };
+	};
 }
 
-const MatchView: React.FC<Props> = ({ reportWin, reportContested, match, loading }) => {
+const MatchView: React.FC<Props> = ({
+	reportWin,
+	reportContested,
+	reportFinalized,
+	match,
+	loading,
+	tournament,
+}) => {
+	function firstTeamRetcon() {
+		reportWin(match.firstTeam.id);
+		console.log(reportFinalized());
+	}
+	function secondTeamRetcon() {
+		reportWin(match.secondTeam.id);
+		console.log(reportFinalized());
+	}
+
 	if (match.winner) {
 		return (
 			<div>
 				<MatchViewItem match={match} />
 				<h1>The winner is: {match.winner.name}</h1>
-				{match.contested && <h1>The match has been contested!</h1>}
+				{match.contested && !match.finalized && <h1>The match has been contested!</h1>}
+				{match.contested && !match.finalized && (
+					<AdminCheck ownerId={tournament.owner.id}>
```

Comment:
Vems ownerId är det första? ÄR det usern som clickar isåfall? Fattar att den andra är det ID som står reggat på turneringen.

Nu blev de svenska för enklare förklaring menmen

Reply:
The first "ownerId" in <AdminCheck ownerId={tournament.owner.id}> is referencing the field name in the AdminCheck component. AdminCheck has an interface that expects a field named "ownerId" of type string. This is only a field name and could be anything. The second one is just from the passed object.